<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
<meta http-equiv="x-ua-compatible" content="IE=edge"/>
<title>Test results - Class itrs_appserver.UserControllerTest</title>
<link href="../css/base-style.css" rel="stylesheet" type="text/css"/>
<link href="../css/style.css" rel="stylesheet" type="text/css"/>
<script src="../js/report.js" type="text/javascript"></script>
</head>
<body>
<div id="content">
<h1>Class itrs_appserver.UserControllerTest</h1>
<div class="breadcrumbs">
<a href="../index.html">all</a> &gt; 
<a href="../packages/itrs_appserver.html">itrs_appserver</a> &gt; UserControllerTest</div>
<div id="summary">
<table>
<tr>
<td>
<div class="summaryGroup">
<table>
<tr>
<td>
<div class="infoBox" id="tests">
<div class="counter">11</div>
<p>tests</p>
</div>
</td>
<td>
<div class="infoBox" id="failures">
<div class="counter">0</div>
<p>failures</p>
</div>
</td>
<td>
<div class="infoBox" id="ignored">
<div class="counter">0</div>
<p>ignored</p>
</div>
</td>
<td>
<div class="infoBox" id="duration">
<div class="counter">1.443s</div>
<p>duration</p>
</div>
</td>
</tr>
</table>
</div>
</td>
<td>
<div class="infoBox success" id="successRate">
<div class="percent">100%</div>
<p>successful</p>
</div>
</td>
</tr>
</table>
</div>
<div id="tabs">
<ul class="tabLinks">
<li>
<a href="#tab0">Tests</a>
</li>
<li>
<a href="#tab1">Standard output</a>
</li>
<li>
<a href="#tab2">Standard error</a>
</li>
</ul>
<div id="tab0" class="tab">
<h2>Tests</h2>
<table>
<thead>
<tr>
<th>Test</th>
<th>Duration</th>
<th>Result</th>
</tr>
</thead>
<tr>
<td class="success">checkLoggedInStatusTest</td>
<td>0s</td>
<td class="success">passed</td>
</tr>
<tr>
<td class="success">loginAuthotrisedDomainFirstTimeTest</td>
<td>0.354s</td>
<td class="success">passed</td>
</tr>
<tr>
<td class="success">loginAuthotrisedDomainSecondTimeWithSecondDeviceTest</td>
<td>0.596s</td>
<td class="success">passed</td>
</tr>
<tr>
<td class="success">loginAuthotrisedDomainSecondTimeWithoutVeriticationTest</td>
<td>0.303s</td>
<td class="success">passed</td>
</tr>
<tr>
<td class="success">loginAuthotrisedUserWithFirstDeviceTest</td>
<td>0.036s</td>
<td class="success">passed</td>
</tr>
<tr>
<td class="success">loginAuthotrisedUserWithSecondDeviceTest</td>
<td>0.046s</td>
<td class="success">passed</td>
</tr>
<tr>
<td class="success">loginFailureCatchTest</td>
<td>0.009s</td>
<td class="success">passed</td>
</tr>
<tr>
<td class="success">loginUnauthorisedDomainTest</td>
<td>0.014s</td>
<td class="success">passed</td>
</tr>
<tr>
<td class="success">logoutTest</td>
<td>0.047s</td>
<td class="success">passed</td>
</tr>
<tr>
<td class="success">setCustomDVTestFailTest</td>
<td>0.001s</td>
<td class="success">passed</td>
</tr>
<tr>
<td class="success">setCustomDVTestSucceedTest</td>
<td>0.037s</td>
<td class="success">passed</td>
</tr>
</table>
</div>
<div id="tab1" class="tab">
<h2>Standard output</h2>
<span class="code">
<pre>10:09:23.042 [OpenAccessApiClient-UnknownDevBuild-akka.actor.default-dispatcher-8] INFO  c.itrsgroup.actor.MessageDispatcher - Authenticating with remote service [/user/frontEnd] at [192.168.220.54:2551]
10:09:23.242 [OpenAccessApiClient-UnknownDevBuild-akka.actor.default-dispatcher-15] INFO  c.itrsgroup.actor.MessageDispatcher - Successfully connected to remote service [akka.tcp://ClusterSystem@192.168.220.54:2551/user/frontEnd]
10:09:23.243 [OpenAccessApiClient-UnknownDevBuild-akka.actor.default-dispatcher-15] INFO  c.itrsgroup.actor.MessageDispatcher - Forwarding Subscription op to cluster frontend: Subscribe(ModelSub(/geneos/gateway[(@name=&quot;GW_TEST_50144&quot;)]/directory/probe[(@name=&quot;SYSMON&quot;)]/managedEntity[(@name=&quot;Android&quot;)]/sampler[(@name=&quot;AppTest&quot;)][(@type=&quot;&quot;)]/dataview[(@name=&quot;AppTest&quot;)]/rows/row[wild(@name,&quot;*&quot;)]/cell,Set(),LIST,),None)
10:09:23.370 [OpenAccessApiClient-UnknownDevBuild-akka.actor.default-dispatcher-18] INFO  c.itrsgroup.actor.MessageDispatcher - Successfully connected to remote service [akka.tcp://ClusterSystem@192.168.220.54:2551/user/frontEnd]
10:09:23.370 [OpenAccessApiClient-UnknownDevBuild-akka.actor.default-dispatcher-18] INFO  c.itrsgroup.actor.MessageDispatcher - Forwarding Subscription op to cluster frontend: Subscribe(ModelSub(testb3/rows/row[wild(@name,&quot;*&quot;)]/cell,Set(),LIST,),None)
10:09:23.540 [OpenAccessApiClient-UnknownDevBuild-akka.actor.default-dispatcher-8] INFO  c.itrsgroup.actor.MessageDispatcher - Successfully connected to remote service [akka.tcp://ClusterSystem@192.168.220.54:2551/user/frontEnd]
10:09:23.540 [OpenAccessApiClient-UnknownDevBuild-akka.actor.default-dispatcher-8] INFO  c.itrsgroup.actor.MessageDispatcher - Forwarding Subscription op to cluster frontend: Subscribe(ModelSub(testa3/rows/row[wild(@name,&quot;*&quot;)]/cell,Set(),LIST,),None)
10:09:23.773 [OpenAccessApiClient-UnknownDevBuild-akka.actor.default-dispatcher-20] INFO  c.itrsgroup.actor.MessageDispatcher - Successfully connected to remote service [akka.tcp://ClusterSystem@192.168.220.54:2551/user/frontEnd]
10:09:23.773 [OpenAccessApiClient-UnknownDevBuild-akka.actor.default-dispatcher-20] INFO  c.itrsgroup.actor.MessageDispatcher - Forwarding Subscription op to cluster frontend: Subscribe(ModelSub(testi3/rows/row[wild(@name,&quot;*&quot;)]/cell,Set(),LIST,),None)
10:09:23.973 [OpenAccessApiClient-UnknownDevBuild-akka.actor.default-dispatcher-10] INFO  c.itrsgroup.actor.MessageDispatcher - Successfully connected to remote service [akka.tcp://ClusterSystem@192.168.220.54:2551/user/frontEnd]
10:09:23.973 [OpenAccessApiClient-UnknownDevBuild-akka.actor.default-dispatcher-10] INFO  c.itrsgroup.actor.MessageDispatcher - Forwarding Subscription op to cluster frontend: Subscribe(ModelSub(teste3/rows/row[wild(@name,&quot;*&quot;)]/cell,Set(),LIST,),None)
This is the thread xpath : test
10:09:24.066 [OpenAccessApiClient-UnknownDevBuild-akka.actor.default-dispatcher-5] INFO  akka.event.slf4j.Slf4jLogger - Slf4jLogger started
10:09:24.070 [OpenAccessApiClient-UnknownDevBuild-akka.actor.default-dispatcher-6] INFO  Remoting - Starting remoting
10:09:24.085 [OpenAccessApiClient-UnknownDevBuild-akka.actor.default-dispatcher-5] INFO  Remoting - Remoting started; listening on addresses :[akka.tcp://OpenAccessApiClient-UnknownDevBuild@192.168.10.207:21791]
[teste]
10:09:24.086 [OpenAccessApiClient-UnknownDevBuild-akka.actor.default-dispatcher-2] INFO  c.itrsgroup.actor.MessageDispatcher - Attempting to connect to remote service at [192.168.220.54:2551] 1 time(s)
10:09:24.141 [OpenAccessApiClient-UnknownDevBuild-akka.actor.default-dispatcher-9] INFO  akka.contrib.pattern.ClusterClient - Connected to [akka.tcp://ClusterSystem@192.168.220.54:2551/user/receptionist]
10:09:24.154 [OpenAccessApiClient-UnknownDevBuild-akka.actor.default-dispatcher-2] INFO  c.itrsgroup.actor.MessageDispatcher - Authenticating with remote service [/user/frontEnd] at [192.168.220.54:2551]
10:09:24.267 [OpenAccessApiClient-UnknownDevBuild-akka.actor.default-dispatcher-3] INFO  c.itrsgroup.actor.MessageDispatcher - Successfully connected to remote service [akka.tcp://ClusterSystem@192.168.220.54:2551/user/frontEnd]
10:09:24.267 [OpenAccessApiClient-UnknownDevBuild-akka.actor.default-dispatcher-3] INFO  c.itrsgroup.actor.MessageDispatcher - Forwarding Subscription op to cluster frontend: Subscribe(ModelSub(testg3/rows/row[wild(@name,&quot;*&quot;)]/cell,Set(),LIST,),None)
10:09:24.414 [OpenAccessApiClient-UnknownDevBuild-akka.actor.default-dispatcher-6] INFO  c.itrsgroup.actor.MessageDispatcher - Successfully connected to remote service [akka.tcp://ClusterSystem@192.168.220.54:2551/user/frontEnd]
10:09:24.414 [OpenAccessApiClient-UnknownDevBuild-akka.actor.default-dispatcher-6] INFO  c.itrsgroup.actor.MessageDispatcher - Forwarding Subscription op to cluster frontend: Subscribe(ModelSub(testa10/rows/row[wild(@name,&quot;*&quot;)]/cell,Set(),LIST,),None)
</pre>
</span>
</div>
<div id="tab2" class="tab">
<h2>Standard error</h2>
<span class="code">
<pre>Feb 04, 2016 10:09:23 AM geneos_notification.loggers.LogObject doLog
INFO: Login attempt with username : testb@Default
Feb 04, 2016 10:09:23 AM geneos_notification.loggers.LogObject doLog
INFO: [SQL]Checking user : testb@Default &amp; testb
Feb 04, 2016 10:09:23 AM geneos_notification.loggers.LogObject doLog
INFO: New user testb@Default has attempted to login, verification email sent and account is in probation.
Feb 04, 2016 10:09:23 AM geneos_notification.loggers.LogObject doLog
INFO: Login attempt with username : testb@Default
Feb 04, 2016 10:09:23 AM geneos_notification.loggers.LogObject doLog
INFO: [SQL]Checking user : testb@Default &amp; testz
Feb 04, 2016 10:09:23 AM geneos_notification.loggers.LogObject doLog
INFO: [SQL]Device Check = User has yet to verify device : testz
Feb 04, 2016 10:09:23 AM geneos_notification.loggers.LogObject doLog
INFO: testb@Default has been sent a verification email for testz
Feb 04, 2016 10:09:23 AM geneos_notification.loggers.LogObject doLog
INFO: Login attempt with username : test@Default
Feb 04, 2016 10:09:23 AM geneos_notification.loggers.LogObject doLog
INFO: [SQL]Checking user : test@Default &amp; test
Feb 04, 2016 10:09:23 AM geneos_notification.loggers.LogObject doLog
INFO: [T-INFO]Thread Execution successful
Feb 04, 2016 10:09:23 AM geneos_notification.loggers.LogObject doLog
WARNING: [T-WARNING]OpenAccess null response due to multiple reloads, this is handled and expected.
Feb 04, 2016 10:09:23 AM geneos_notification.loggers.LogObject doLog
INFO: [T-INFO]Thread Execution successful
Feb 04, 2016 10:09:23 AM geneos_notification.loggers.LogObject doLog
WARNING: [T-WARNING]OpenAccess null response due to multiple reloads, this is handled and expected.
Feb 04, 2016 10:09:23 AM geneos_notification.loggers.LogObject doLog
INFO: New user test@Default has attempted to login, verification email sent and account is in probation.
Feb 04, 2016 10:09:23 AM geneos_notification.loggers.LogObject doLog
INFO: Login attempt with username : fail' or 1 = 1
Feb 04, 2016 10:09:23 AM geneos_notification.loggers.LogObject doLog
INFO: [SQL]Checking user : fail' or 1 = 1 &amp; fail' or 1 = 1
Feb 04, 2016 10:09:23 AM geneos_notification.loggers.LogObject doLog
SEVERE: [SQL]SQL query issue was encountered causing SQL failure : com.mysql.jdbc.exceptions.MySQLSyntaxErrorException: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ''' at line 1
java.lang.RuntimeException: com.mysql.jdbc.exceptions.MySQLSyntaxErrorException: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ''' at line 1
	at geneos_notification.controllers.DatabaseController.checkUser(DatabaseController.java:75)
	at geneos_notification.controllers.UserController.login(UserController.java:27)
	at itrs_appserver.UserControllerTest.loginFailureCatchTest(UserControllerTest.java:102)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecuter.runTestClass(JUnitTestClassExecuter.java:86)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecuter.execute(JUnitTestClassExecuter.java:49)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassProcessor.processTestClass(JUnitTestClassProcessor.java:64)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:50)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.messaging.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:32)
	at org.gradle.messaging.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93)
	at com.sun.proxy.$Proxy2.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass(TestWorker.java:106)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.messaging.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:360)
	at org.gradle.internal.concurrent.DefaultExecutorFactory$StoppableExecutorImpl$1.run(DefaultExecutorFactory.java:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: com.mysql.jdbc.exceptions.MySQLSyntaxErrorException: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ''' at line 1
	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:936)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:2985)
	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:1631)
	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:1723)
	at com.mysql.jdbc.Connection.execSQL(Connection.java:3277)
	at com.mysql.jdbc.Connection.execSQL(Connection.java:3206)
	at com.mysql.jdbc.Statement.executeQuery(Statement.java:1232)
	at geneos_notification.controllers.DatabaseController.checkUser(DatabaseController.java:58)
	... 46 more
Feb 04, 2016 10:09:23 AM geneos_notification.loggers.LogObject doLog
SEVERE: Login error has occured, please see associated stack trace for more information. Username : fail' or 1 = 1 android_id : fail' or 1 = 1
Feb 04, 2016 10:09:23 AM geneos_notification.loggers.LogObject doLog
INFO: java.lang.RuntimeException: com.mysql.jdbc.exceptions.MySQLSyntaxErrorException: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ''' at line 1
Feb 04, 2016 10:09:23 AM geneos_notification.loggers.LogObject doLog
INFO: Login attempt with username : testf@Default
Feb 04, 2016 10:09:23 AM geneos_notification.loggers.LogObject doLog
INFO: [SQL]Checking user : testf@Default &amp; testf
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: [SQL]Device testf logged in successfully
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: Initial log on for testf@Default with device testf
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: [SQL]Device testf logged out successfully
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: The status of the device check fortestf@Default is :1
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: testf@Default has been successfully logged out.
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: Login attempt with username : teste@Default
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: [SQL]Checking user : teste@Default &amp; teste
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: [SQL]Device teste logged in successfully
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: Initial log on for teste@Default with device teste
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: [T-INFO]Initialization of thread for xpath : test
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: Login attempt with username : something@wrong
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: [SQL]Checking user : something@wrong &amp; test
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
WARNING: [SQL]User has tried to log in with an invalid domain : something@wrong
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
WARNING: Attempted login made by unauthorised user with username : something@wrong
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: Thread xpath exection for path : test
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: Login attempt with username : testc@Default
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: [SQL]Checking user : testc@Default &amp; testc
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: [SQL]Device testc logged in successfully
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: Initial log on for testc@Default with device testc
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
SEVERE: Attempted access by invalid and potentially harmful user utilising username : testFail@Default
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: Login attempt with username : testd@Default
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: [SQL]Checking user : testd@Default &amp; testd
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: [SQL]Device testd logged in successfully
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: Initial log on for testd@Default with device testd
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: Login attempt with username : testd@Default
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: [SQL]Checking user : testd@Default &amp; testx
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: [SQL]Device testx logged in successfully
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: testd@Default logged in successfully on device testx
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: Login attempt with username : testa@Default
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: [SQL]Checking user : testa@Default &amp; testa
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: New user testa@Default has attempted to login, verification email sent and account is in probation.
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: Login attempt with username : testa@Default
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: [SQL]Checking user : testa@Default &amp; testa
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: [SQL]Device Check = User has yet to verify device : testa
Feb 04, 2016 10:09:24 AM geneos_notification.loggers.LogObject doLog
INFO: testa@Default has already been sent a verification email for testa
</pre>
</span>
</div>
</div>
<div id="footer">
<p>
<div>
<label class="hidden" id="label-for-line-wrapping-toggle" for="line-wrapping-toggle">Wrap lines
<input id="line-wrapping-toggle" type="checkbox" autocomplete="off"/>
</label>
</div>Generated by 
<a href="http://www.gradle.org">Gradle 2.3</a> at 04-Feb-2016 10:09:25</p>
</div>
</div>
</body>
</html>
